cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowHttpHandler.java

```diff
@@ -323,4 +341,85 @@ private WorkflowToken getWorkflowToken(String namespaceId, String appName, Strin
     }
     return store.getWorkflowToken(workflowId, runId);
   }
+
+  @GET
+  @Path("/apps/{app-id}/workflows/{workflow-id}/runs/{run-id}/localdatasets")
+  public void getWorkflowLocalDatasets(HttpRequest request, HttpResponder responder,
+                                       @PathParam("namespace-id") String namespaceId,
+                                       @PathParam("app-id") String applicationId,
+                                       @PathParam("workflow-id") String workflowId,
+                                       @PathParam("run-id") String runId)
+    throws NotFoundException, DatasetManagementException {
+    WorkflowSpecification workflowSpec = getWorkflowSpecForValidRun(namespaceId, applicationId, workflowId, runId);
+    Map<String, DatasetCreationSpec> localDatasets = new HashMap<>();
```

Comment:
the regular datasets endpoint (/v3/namespaces/ns/data/datasets) returns a list of DatasetSpecificationSummary. I think it would be better to return that. The only difference is it uses name and type as field names instead of instanceName and typeName.